### PR TITLE
Added a new option to allow for isolating the denotation character

### PR DESCRIPTION
**Pull request by Datedsandwich**
This library is almost perfect for our use case, but we noticed that email addresses trigger the mention list when the denotation character is '@', so we've made a small change to allow a new option that won't trigger the mention list unless the denotation character is isolated.

### CODE REVIEW

**Comment by MadSpindel** [comment]
It's hard to follow the changes made in this pull request because of the formatting. Could you make a new pull request with only the changes?

**Comment by Datedsandwich** [comment]
There was a second commit that undid the formatting changes
